### ISSUE TITLE
Docstrings should not affect test display

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#1897]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#1897|
|Type|defect|
|Created|2006-07-04 10:58:12Z|

At the moment, Trial follows unittest's lead and changes the display of tests (in verbose and tree modes) if the first line of the docstring is set.

This is actually lousy.  We can change it, and it will cause no incompatibility with pyunit -- just a difference of output intended only for human viewing.

<details><summary>Searchable metadata</summary>

```
trac-id__1897 1897
type__defect defect
reporter__jml jml
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1152010692000000 1152010692000000
changetime__1152421009000000 1152421009000000
version__None None
owner__ 
cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Hooray

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv commented|
|-|-|

Yes please!

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @exarkun<br>@jml set status to `new`|
|-|-|

See [branches/docstring-1897](branches/docstring-1897)

Sorry to have to assign to you again, Jp...

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @jml|
|-|-|

Thank you!  This behavior has annoyed me for a long time.

Nice and simple.  Could you add a docstring for getDescription, though?

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set status to `closed`|
|-|-|

(In [17510]) Docstrings of test cases no longer affect Trial's output

 * Author: jml
 * Reviewer: exarkun
 * Fixes [#1897](https://github.com/twisted/twisted/issues/1897)

Always display the name of the test method in TreeReporter.
Even if the test method has a docstring.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
